src/components/Signup.js

```diff
@@ -0,0 +1,216 @@
+import React from "react";
+
+import { connect } from 'react-redux';
+import {setUserDetails} from '../actions/signupActions';
+import {resetCartData} from '../actions/productActions';
+import {validEmail,validPwd} from '../utils';
+import Header from '../presentations/Header';
+import Footer from '../presentations/Footer';
+import toastr from 'toastr';
+import { Modal } from "react-responsive-modal";
+import Cart from '../presentations/Cart';
+import Constants from '../constants';
+class Signup extends React.Component {
+    constructor(props){
+        super(props);
+        this.state={
+            data:{
+                email:'',
+                password:'',
+                confirmPassword:'',
+                fName:'',
+                lName:'',
+                formError: false
+              }
+        }
+        // this.onSubmitHandler = this.onSubmitHandler.bind(this);
+        // this.inputHandler = this.inputHandler.bind(this);
+        // this.isValidFormState = this.isValidFormState.bind(this);
+        // this.showCartView = this.showCartView.bind(this);
+        // this.onCloseModal = this.onCloseModal.bind(this);
+        // this.checkoutComplete = this.checkoutComplete.bind(this);
+    }
+
+    static getDerivedStateFromProps(props,state){
+        if(props.signUp && props.signUp.userData){
+            let title = `${Constants.TEXTS.DEFAULTS.hai}, ${state.fName}`
+            let message = Constants.TEXTS.SIGNUP.signup_success;
+            toastr.success(message, title,{timeOut:1000});
+            props.router.push('/Home');
+        }
+        return state;
```

Comment:
why return state if it is not changed.
is using getDerived StateFromProps even necessary for the usecase you have?